### PR TITLE
Migrate to GH Actions

### LINKED ISSUE #101: Migrate to GitHub Actions
travis-ci.org is going to be shutdown in December so we'll need to migrate off before then. Since open-policy-agent/{conftest, gatekeeper, opa} have been using GHA successfully for the past few months, we should migrate frameworks ASAP.

### PR DESCRIPTION
**Pull request by sozercan**
Signed-off-by: Sertac Ozercan <sozercan@gmail.com>

Fixes #101

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,4 +2,4 @@
 all: build-constraint
 
 build-constraint:
-	./build/build_constraint.sh
+	./build/build_constraint.sh
```
Comment (maxsmythe) [inline Makefile:5]: nit: newline

File: constraint/Makefile [hunk 2]
```diff
@@ -33,16 +33,8 @@ manifests: buildutils
...
-# Run go fmt against code
-fmt:
-	go fmt ./pkg/...
-
-# Run go vet against code
-vet:
-	go vet ./pkg/...
-
 lint:
-	golangci-lint -v run ./... --timeout 5m --skip-files '(^|/)zz_generated'
```
Comment (maxsmythe) [inline constraint/Makefile:45]: do the generated files now comply with the linter?

File: constraint/deploy/crds.yaml [hunk 3]
```diff
@@ -17,14 +17,10 @@ spec:
     openAPIV3Schema:
       properties:
         apiVersion:
-          description: 'APIVersion defines the versioned schema of this representation
-            of an object. Servers should convert recognized schemas to the latest
-            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+          description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
```
Comment (maxsmythe) [inline constraint/deploy/crds.yaml:20]: For some reason you are generating non-wrapped manifests?

**Review by maxsmythe — COMMENTED** [review]
Mostly LGTM, some nits, sorry for the delay.

File: Makefile (hunk 1 above)
Comment (sozercan) [reply Makefile]: updated

File: constraint/Makefile (hunk 2 above)
Comment (sozercan) [reply constraint/Makefile:45]: seems fine https://github.com/sozercan/frameworks/runs/1383770406?check_suite_focus=true

File: constraint/deploy/crds.yaml (hunk 3 above)
Comment (sozercan) [reply constraint/deploy/crds.yaml:20]: This repo has controller-gen v0.1.8, that's probably the reason we are seeing non-wrapped lines

Comment (sozercan) [reply constraint/deploy/crds.yaml:20]: looks like this is due to kustomize version, repo uses v3.0.2 and i have v3.7.0 locally

Comment (maxsmythe) [reply constraint/deploy/crds.yaml]: Interesting, I wonder why Kustomize version affects this?

**Review by maxsmythe — APPROVED** [review]
LGTM

File: constraint/deploy/crds.yaml (hunk 3 above)
Comment (maxsmythe) [reply constraint/deploy/crds.yaml]: NVM, I was thinking kubebuilder, not kustomize

Comment (maxsmythe) [reply constraint/deploy/crds.yaml]: @brycecr @juliankatz

It looks like version skew of kustomize is what's causing the different wrappings for the manifest. Do we want to Dockerize kustomize as well?

**Pull request merged**